api/src/test/java/org/openmrs/api/impl/ConditionServiceImplTest.java

```diff
@@ -0,0 +1,180 @@
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.api.impl;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.openmrs.CodedOrFreeText;
+import org.openmrs.Condition;
+import org.openmrs.ConditionClinicalStatus;
+import org.openmrs.ConditionVerificationStatus;
+import org.openmrs.api.ConditionService;
+import org.openmrs.api.PatientService;
+import org.openmrs.api.context.Context;
+import org.openmrs.test.BaseContextSensitiveTest;
+import java.util.Date;
+import java.util.List;
+
+/**
+ * Unit tests for methods that are specific to the {@link ConditionServiceImpl}. General tests that
+ * would span implementations should go on the {@link ConditionService}.
+ */
+public class ConditionServiceImplTest extends BaseContextSensitiveTest {
+	
+	protected static final String CONDITION_XML = "org/openmrs/api/include/ConditionServiceImplTest-SetupCondition.xml";
+	
+	private ConditionService conditionService;
+	
+	private PatientService patientService;
+	
+	
+	@Before
+	public void setup (){
+		if(conditionService == null){
+			conditionService = Context.getConditionService();
+		}
+		if(patientService == null){
+			patientService = Context.getPatientService();
+		}
+		executeDataSet(CONDITION_XML);
+	}
+
+	/**
+	 * @see ConditionService#saveCondition(Condition) 
+	 */
+	@Test
+	public void saveCondition_shouldSaveNewCondition(){
+		Integer patientId = 2;
+		String uuid = "08002000-4469-12q3-551f-0339000c9a76";
+		CodedOrFreeText codedOrFreeText = new CodedOrFreeText();
+		Condition condition = new Condition();
+		condition.setCondition(codedOrFreeText);
+		condition.setClinicalStatus(ConditionClinicalStatus.ACTIVE);
+		condition.setUuid(uuid);
+		condition.setPatient(patientService.getPatient(patientId));
+		conditionService.saveCondition(condition);
+		Condition savedCondition = conditionService.getConditionByUuid(uuid);
+		Assert.assertNotNull(savedCondition.getConditionId());
+	}
+
+	/**
+	 * @see ConditionService#getConditionByUuid(String)
+	 */
+	@Test
+	public void getConditionByUuid_shouldFindConditionGivenValidUuid() {
+		Integer conditionId = 1;
+		String uuid="2cc6880e-2c46-15e4-9038-a6c5e4d22fb7";
+		Condition condition = conditionService.getConditionByUuid(uuid);
+		Assert.assertEquals(conditionId, condition.getConditionId());
+	}
+
+	/**
+	 * @see ConditionService#getConditionByUuid(String)
+	 */
+	@Test
+	public void getConditionByUuid_shouldReturnNullIfNoObjectFoundWithGivenUuid() {
+		Assert.assertNull(conditionService.getConditionByUuid("invalid uuid"));
+	}
+
+	/**
+	 * @see ConditionService#getCondition(Integer) 
+	 */
+	@Test
+	public void getCondition_shouldFindConditionGivenValidId() {
+		int conditionId = 1;
+		String uuid="2cc6880e-2c46-15e4-9038-a6c5e4d22fb7";
```

Comment:
If you are using these variable once, then there is not need for them. Just directly use the values where expected

Reply:
Okay. Noted